src/renderer/geometry/instanced_mesh.rs

```diff
@@ -379,12 +359,8 @@ impl Geometry for InstancedMesh {
 ///
 #[derive(Clone, Debug, Default)]
 pub struct Instances {
-    /// The translation applied to the positions of each instance.
-    pub translations: Vec<Vec3>,
-    /// The rotations applied to the positions of each instance.
-    pub rotations: Option<Vec<Quat>>,
-    /// The non-uniform scales applied to the positions of each instance.
-    pub scales: Option<Vec<Vec3>>,
+    /// The transformations applied to each instance.
+    pub transformations: Vec<Mat4>,
```

Comment:
Massive bikeshedding opportunity here... I actually ran into the "can't extract scale from Mat4" issue this evening so switched to this branch, and I instinctively wrote `transforms`, not `transformations`.

The [current](https://docs.rs/three-d/latest/three_d/renderer/geometry/struct.Instances.html) `Instances` object has `texture_transforms` not `texture_transformations`, should we make that consistent while we are modifying this structure anyway?

The internals of `InstancedMesh` use [both](https://github.com/asny/three-d/blob/966a4ede78f0314c94291d35e162ed2201e84482/src/renderer/geometry/instanced_mesh.rs#L15-L16) flavours. I don't think there's currently an established precedence for this word in the public API.

I'm in favour of `transforms` over `transformations`, just because it is more concise and easier to type.

Reply:
It's funny, I actually looked up the difference between transformation and transform when I decided on the naming, and they are used interchangeably but usually transform is used in relation to a specific transformation, for example the Laplace transform. So I went with transformation and that should be consistent across the crate (if it's not, please let me know). However, I'm not sure the texture transform is a specific transformation or just a transformation? I think I called it transform to be consistent with three js or unity or something. Naming is hard! 😬

Reply:
Yeah, I would use the `transformation` describes the actual value, and `transform` is the verb that applies a `transformation`.

> So I went with transformation and that should be consistent across the crate (if it's not, please let me know)

In InstancedMesh, linked in the previous comment the `transformation` and `instance_transforms` private fields definitely are the same use of the word, both describe the actual value. 

> I think I called it transform to be consistent with three js or unity or something. Naming is hard!

Yep, naming is definitely hard, I find `transform` easier to type than `transformation`. And I think using `transform` is indeed more in line with other libraries. [Unreal Engine](https://docs.unrealengine.com/4.26/en-US/BlueprintAPI/Math/Transform/MakeRelativeTransform/), [Unity](https://docs.unity3d.com/ScriptReference/Transform.html), [Bevy](https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html) and even [cgmath::Transform](https://docs.rs/cgmath/latest/cgmath/trait.Transform.html). On the `Transform` trait they have `transform_vector()` (applies the transformation to a point) as well as `inverse_transform()` (returns the inverted value).

I think this is the first point in the API where it's a public facing decision instead of an internal variable naming one. I'd go with `transform`, `cgmath` already uses that, and it is also in line with other 3d frameworks / game engines.

Reply:
I'm not sure I'm following what your preferred solution is 🙂 I get the noun verb thing, but that just confirms that it should be transformation. There's also `set_transformation()` methods which you could argue could be renamed to `transform()`. However, the main question as I see it is whether it should be `texture_transform` or `texture_transformation`.

Reply:
Haha, sorry xD . Yes, I think in the most correct sense it should be `transformation` (and `texture_transformation`). But I would argue for using `transform` because it's shorter and more commonly used.

Reply:
I think then I have to stick with `transformation`, it's not a good enough reason to break backwards compatibility, but if it was a new feature, I wouldn't mind calling it `transform`. However, I will rename to `texture_transformation` in #298 . Is that ok? 🙂 And then I will merge this 💪